core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -484,9 +489,11 @@ private[kafka] class Processor(val id: Int,
   protected[network] def sendResponse(response: RequestChannel.Response, responseSend: Send) {
     val connectionId = response.request.connectionId
     trace(s"Socket server received response to send to $connectionId, registering for write and sending data: $response")
-    val channel = selector.channel(connectionId)
+    // Invoke send for closingChannel as well so that the send is failed and the channel closed properly after
+    // discarding any pending staged receives.
```

Comment:
Reinstated the warning for `closingChannels`.

No, we add to `disconnected` only when the channel is removed from `closingChannels`. So this triggers the only addition to `disconnected`.